rs/nervous_system/common/test_utils/src/wasm_helpers.rs

```diff
@@ -0,0 +1,36 @@
+use ic_wasm;
+use libflate::gzip;
+
+/// A small, valid WASM suitable for tests.
+pub const SMALLEST_VALID_WASM_BYTES: &[u8; 8] = &[0, 0x61, 0x73, 0x6D, 1, 0, 0, 0];
+
+/// Return a version of `wasm` extended with a metadata section
+/// "icp:[public|private] $name$contents".
+pub fn annotate_wasm_with_metadata(
+    wasm: &[u8],
+    is_public: bool,
+    name: &str,
+    contents: Vec<u8>,
+) -> Vec<u8> {
+    use ic_wasm::{metadata, utils};
+
+    let kind = if is_public {
+        metadata::Kind::Public
+    } else {
+        metadata::Kind::Private
+    };
+
+    let mut wasm_module = utils::parse_wasm(wasm, false).expect("Cannot parse WASM");
+    metadata::add_metadata(&mut wasm_module, kind, name, contents);
+    wasm_module.emit_wasm()
+}
+
+// Gzips a wasm, returning the hash of its compressed representation.
+pub fn gzip_wasm(wasm: &[u8]) -> Vec<u8> {
```

Comment:
I think there's a function somewhere called something like "ensure_gzipped" so that you can pass it bytes that are gzipped or not, and either zip it or leave it alone depending.

Reply:
You're probably referring to the `ensure_sns_wasm_gzipped` function which has the following signature:

```
pub fn ensure_sns_wasm_gzipped(mut sns_wasm: SnsWasm) -> SnsWasm
```

I wanated to have a low-level version of that that doesn't refer to the API type `SnsWasm`. 

As a side remark, ic-wasm that will be used in production for SNS-W uses libflate rather that flate2 (both seem like legit libraries, from a quick google).